ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,252 @@
+      partitionIds.add(partition);
+      partition2Replicas.put(partition, replica);
+    }
+  }
+
+  public void run() {
+    List<RequestInfo> requestInfos = null;
+
+    while (!shutdown) {
+      try {
+        // sleep for some time
+        Thread.sleep(SLEEP_TIME_MS);
+
+        boolean hasMore = true;
+        boolean hasError = false;
+        // if some partitions have more requests, continue to handle them.
+        while (hasMore && !hasError) {
+          hasMore = false;
+          hasError = false;
+          // loop all the partitions. For each partition, we handle maxResults number of requests.
+          for (long partitionId : partitionIds) {
+            if (shutdown) {
+              return;
+            }
+            // get the repair requests for this partition
+            requestInfos = getAmbryRequest(partitionId);
+            if (requestInfos.isEmpty()) {
+              continue;
+            }
+            // this partition may have more requests.
+            if (requestInfos.size() >= maxResults) {
+              hasMore = true;
+            }
+
+            Set<Integer> requestsToDrop = new HashSet<>();
+            // send the repair requests to the handler and wait for the responses.
+            while (!requestInfos.isEmpty()) {
+              RequestInfo reqInfo = requestInfos.get(requestInfos.size() - 1);
+              List<ResponseInfo> responses =
+                  client.sendAndPoll(Collections.singletonList(reqInfo), requestsToDrop, POLL_TIMEOUT_MS);
+              if (responses == null || responses.size() == 0) {
+                logger.error("RepairRequests Sender: timeout waiting for repairs {} {}", nodeId, reqInfo.getRequest());
+                hasError = true;
+                break;
+              }
+              ResponseInfo resInfo = responses.get(0);
+              ReplicateBlobResponse res = (ReplicateBlobResponse) resInfo.getResponse();
+              ReplicateBlobRequest req = (ReplicateBlobRequest) resInfo.getRequestInfo().getRequest();
+              if (res.getError() == ServerErrorCode.No_Error) {
+                RepairRequestRecord.OperationType type;
+                if (req.getOperationType() == RequestOrResponseType.TtlUpdateRequest) {
+                  type = RepairRequestRecord.OperationType.TtlUpdateRequest;
+                } else {
+                  type = RepairRequestRecord.OperationType.DeleteRequest;
+                }
+                db.removeRepairRequests(req.getBlobId().toString(), type);
+                logger.info("RepairRequests Sender: Repaired {}", req);
+              } else {
+                // LOCAL_CONSISTENCY_TODO: metrics for all the errors happens in RepairRequestsSender.
+                // Ideally we may need create alert if we cannot repair the requests in timely matter.
+                logger.error("RepairRequests Sender: failed to repair {} response {}", req, res);
+                hasError = true;
```

Comment:
break here?

Reply:
I decided to continue to handle the remaining requests if it returns error status.
Only when the request gets timeout, I'll break the loop.